tests/data/invalid_repetitions.yaml

```diff
@@ -0,0 +1,125 @@
+- input:
+    program:
+      children:
+      - name: a
+        ports:
+        - direction: through
+          name: thru_0
+          size: null
+        resources:
+        - name: y
+          type: additive
+          value: N
+        - name: z
+          type: multiplicative
+          value: N
+        type: null
+      - name: b
+        ports:
+        - direction: through
+          name: thru_0
+          size: null
+        resources:
+        - name: y
+          type: additive
+          value: N
+        - name: z
+          type: multiplicative
+          value: N
+        type: null
+      connections:
+      - source: a.thru_0
+        target: out_0
+      - source: in_0
+        target: a.thru_0
+      - source: b.thru_0
+        target: out_1
+      - source: in_1
+        target: b.thru_0
+      name: root
+      ports:
+      - direction: input
+        name: in_0
+        size: N
+      - direction: output
+        name: out_0
+        size: null
+      - direction: input
+        name: in_1
+        size: N
+      - direction: output
+        name: out_1
+        size: null
+      repetition:
+        count: 10
+        sequence:
+          type: constant
+          multiplier: 1
+      type: null
+    version: v1
+  description: Routine with repetition with more than one child.
+  problems:
+  - "Routine with repetition contains more than one child: ('root',)."
+- input:
+    program:
+      name: root
+      ports:
+      - direction: input
+        name: in_0
+        size: N
+      - direction: output
+        name: out_0
+        size: null
+      repetition:
+        count: 10
+        sequence:
+          type: constant
+          multiplier: 1
+      type: null
+    version: v1
+  description: Routine with repetition without any children.
+  problems:
+  - "Routine with repetition doesn't contain any children: ('root',)."
+- input:
+    program:
+      children:
+      - name: a
+        ports:
+        - direction: through
+          name: thru_0
+          size: null
+        resources:
+        - name: y
+          type: additive
+          value: N
+        - name: z
+          type: multiplicative
+          value: N
+        type: null
+      connections:
+      - source: a.thru_0
+        target: out_0
+      - source: in_0
+        target: a.thru_0
+      name: root
+      ports:
+      - direction: input
+        name: in_0
+        size: N
+      - direction: output
+        name: out_0
+        size: null
+      repetition:
+        count: 10
+        sequence:
+          type: constant
+          multiplier: 1
+      resources:
+      - name: x
+        type: additive
+        value: 10
+      type: null
+    version: v1
+  description: Routine with repetition with resources.
+  problems:
+  - "Routine with repetition should not contain any resources: ('root',), resources: ['x']."
```

Comment:
**nit**: No newline at the end of this file.

Reply:
Done